server/src/main/kotlin/dev/robocode/tankroyale/server/connection/MultiServerWebSocketObserver.kt

```diff
@@ -2,12 +2,20 @@ package dev.robocode.tankroyale.server.dev.robocode.tankroyale.server.connection
 
 import dev.robocode.tankroyale.server.Server
 import org.java_websocket.WebSocket
+import org.slf4j.LoggerFactory
 import java.net.InetAddress
 import java.net.InetSocketAddress
+import kotlin.system.exitProcess
 
 class MultiServerWebSocketObserver(observer: IClientWebSocketObserver) {
 
+    private val log = LoggerFactory.getLogger(this::class.java)
+
     private val webSocketServer = if (Server.useInheritedChannel) {
+        if (System.inheritedChannel() == null) {
+            log.error("'${Server.INHERIT}' (socket activation) is not supported on this system.")
```

Comment:
Just some small feedback on this error message: `System.inheritedChannel()` will be null if starting the server without having passed a socket as fd 3. This doesn't mean this mode of operation is inherently unsupported on this system, just that the user didn't start the application correctly.

I would recommend messaging along the following lines: `'${Server.INHERIT}' can only be used when a socket has been passed as stdin; if you did pass a socket, this feature may be unsupported on this system`

For additional safety, you might want to check that `System.inheritedChannel() instanceof ServerSocketChannel`. If this check fails, but it is `!= null`, then the user did definitely invoke the application incorrectly.